pkg/cli/describe_helm.go

```diff
@@ -0,0 +1,207 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/loft-sh/log"
+	"github.com/loft-sh/vcluster/config"
+	"github.com/loft-sh/vcluster/pkg/cli/find"
+	"github.com/loft-sh/vcluster/pkg/cli/flags"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type DescribeOutput struct {
+	Distro       string   `json:"Distro"`
+	Version      string   `json:"Version"`
+	BackingStore string   `json:"Backing Store"`
+	ImageTags    ImageTag `json:"Image tags"`
+}
+
+type ImageTag struct {
+	APIServer         string `json:"apiServer,omitempty"`
+	Syncer            string `json:"syncer,omitempty"`
+	Scheduler         string `json:"scheduler,omitempty"`
+	ControllerManager string `json:"controllerManger,omitempty"`
+}
+
+func DescribeHelm(ctx context.Context, flags *flags.GlobalFlags, output io.Writer, name, format string) error {
+	namespace := "vcluster-" + name
+	if flags.Namespace != "" {
+		namespace = flags.Namespace
+	}
+
+	secretName := "vc-config-" + name
+
+	kConf := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{})
+	rawConfig, err := kConf.RawConfig()
+	if err != nil {
+		return err
+	}
+	clientConfig, err := kConf.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(ctx, secretName, v1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	configBytes, ok := secret.Data["config.yaml"]
+	if !ok {
+		return fmt.Errorf("secret %s in namespace %s does not contain the expected 'config.yaml' field", secretName, namespace)
+	}
+
+	switch format {
+	case "yaml":
+		_, err = output.Write(configBytes)
+		return err
+	case "json":
+		b, err := yaml.YAMLToJSON(configBytes)
+		if err != nil {
+			return err
+		}
+		_, err = output.Write(b)
+		return err
+	}
+	fVcluster, err := find.GetVCluster(ctx, rawConfig.CurrentContext, name, namespace, log.Discard)
```

Comment:
why not get this earlier?

Reply:
because for the yaml and json output we don't need it